modules/gltf/gltf_document.cpp

```diff
@@ -6858,32 +6881,81 @@ Node *GLTFDocument::generate_scene(Ref<GLTFState> state, int32_t p_bake_fps) {
 			_import_animation(state, ap, i, p_bake_fps);
 		}
 	}
-
+	for (Map<GLTFNodeIndex, Node *>::Element *node_i = state->scene_nodes.front(); node_i; node_i = node_i->next()) {
+		Node *node = state->scene_nodes[node_i->key()];
+		ERR_CONTINUE(!node);
+		for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
+			Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
+			ERR_CONTINUE(ext.is_null());
+			ERR_CONTINUE(!state->json.has("nodes"));
+			Array nodes = state->json["nodes"];
+			ERR_CONTINUE(node_i->key() >= nodes.size());
+			Dictionary node_json = nodes[node_i->key()];
+			Ref<GLTFNode> gltf_node = state->nodes[node_i->key()];
+			err = ext->import_node(state, gltf_node, node_json, node);
+			ERR_CONTINUE(err != OK);
+		}
+	}
 	for (int32_t ext_i = 0; ext_i < document_extensions.size(); ext_i++) {
 		Ref<GLTFDocumentExtension> ext = document_extensions[ext_i];
 		ERR_CONTINUE(ext.is_null());
-		Error err = ext->import_post(this, root);
+		err = ext->import_post(root);
```

Comment:
I think `import_post` may need access to `this`, for example, in order to access the extensions from `gltfState->json`

Reply:
I would like to pretend GLTFDocument does not have state. So I would prefer if the extensions used the set_setting api.

Reply:
I'm sorry. I meant to say it should pass `state` as the first argument. I agree that the GLTFDocument object would be inappropriate.

Reply:
Speaking of which, the functions argument types in `modules/gltf/doc_classes/GLTFDocumentExtension.xml` are out of date, so I may have been confused from that.

Reply:
The second problem is the GLTF state does not have contents at that point.

Reply:
I think it should. It is post, which is after the import process has finished creating and using all the state. The state won't be cleared.